.vitepress/theme/data/types.ts

```diff
@@ -0,0 +1,36 @@
+export type MenuLinkGroup =
+  | 'api'
+  | 'codeflow'
+  | 'teams'
+  | 'enterprise'
+  | 'integrationGuide'
+  | 'userGuide'
+  | 'webcontainers';
+
+export interface MenuLinkItem {
+  text: string;
+  link: string;
+  items?: MenuLinkItem[];
+}
+
+export interface CardLink {
+  bgImg?: 'bolt' | 'squares' | 'steps';
+  description: string;
+  icon?: string;
+  large?: boolean;
+  title: string;
+  url: string;
+}
```

Comment:
This object is used to type some content in `docs/links.ts`, and is identical to the props for the `CardLink.vue` component, but in that component we cannot define the props like this:

```ts
import type { CardLink as CardLinkProps } from '@theme/data/types';

defineProps<CardLinkProps>();
```

Because it breaks the Vue SFC compiler. It's a known issue in Vue. You have to pass at least the prop keys literally:

```ts
defineProps<{
  bgImg?: 'bolt' | 'squares' | 'steps';
  description: string;
  icon?: string;
  large?: boolean;
  title: string;
  url: string;
}>();
```

And you can only used imported types for prop _values_.

So in our case, we end up with duplication.